smithy-model/src/main/java/software/amazon/smithy/model/validation/ValidationEvent.java

```diff
@@ -201,21 +200,14 @@ public static ValidationEvent fromNode(Node node) {
                 objectNode.getNumberMemberOrDefault("line", 0).intValue(),
                 objectNode.getNumberMemberOrDefault("column", 0).intValue());
 
-        // Set required properties.
-        Builder builder = builder()
-                .id(objectNode.expectStringMember("id").getValue())
-                .severity(Severity.valueOf(objectNode.expectStringMember("severity").getValue()))
-                .message(objectNode.expectStringMember("message").getValue())
-                .sourceLocation(location);
-
-        // Set optional properties.
-        objectNode.getStringMember("suppressionReason").map(StringNode::getValue)
-                .ifPresent(builder::suppressionReason);
-        objectNode.getStringMember("shapeId")
-                .map(StringNode::getValue)
-                .map(ShapeId::from)
-                .ifPresent(builder::shapeId);
-
+        Builder builder = builder().sourceLocation(location);
+        objectNode.readObject()
+                .requiredString("id", builder::id)
+                .required("severity", Severity::fromNode, builder::severity)
+                .requiredString("message", builder::message)
+                .requiredString("suppressionReason", builder::suppressionReason)
```

Comment:
Don't believe this should be required based on the previous code.

Reply:
Fixed